cache/task.go

```diff
@@ -0,0 +1,82 @@
+//   Copyright 2019 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package cache
+
+import (
+	"crypto/md5"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/trackit/jsonlog"
+)
+
+func getTotalRedisKeys() (totalKeys int64, err error) {
+	const subStrLen = len("keys=")
+	keySpace := mainClient.Info("keyspace")
+	idx := strings.Index(keySpace.Val(), "keys=")
+	if idx == -1 {
+		return
+	}
+	comma := strings.IndexByte(keySpace.Val()[idx:], ',') - subStrLen
+	if comma == -1 {
+		return
+	}
+	return strconv.ParseInt(keySpace.Val()[idx + subStrLen:idx + subStrLen + comma], 10, 64)
```

Comment:
You should make a comment explaining the "magic" number and indexes that are in your string manipulation